### PR TITLE
chore: updates stories and e2e tests

### LINKED ISSUE #25132: Bug: Dialog Docs nits
### Library

React Components / v9 (@fluentui/react-components)

- [x] [With Form](https://react.fluentui.dev/?path=/docs/preview-components-dialog-dialog--default#with-form) says 'Having a formulary', which I assume should be 'Having a form'.
- [x] [Trigger Outside of Dialog](https://react.fluentui.dev/?path=/docs/preview-components-dialog-dialog--default#trigger-outside-dialog) It appears that the only thing added by DialogTrigger is the aria-haspopup="dialog". If that's the only thing added, we should note that, and not make the use of DialogTrigger sound so critical.  https://github.com/microsoft/fluentui/issues/25132#issuecomment-1274366111
- [x] opening the "Open non-modal dialog" in [No Focusable Element](https://react.fluentui.dev/?path=/docs/preview-components-dialog-dialog--default#no-focusable-element) leaves the user in a state where they have to refresh the page to return to a useable page state. While I understand that this is the lesson we're trying to teach, I feel this is a bit too much for someone that clicked the button out of curiosity. Either provide a way to close the modal, or at least warn what is about to happen. https://github.com/microsoft/fluentui/issues/25150
- [x] We tell people not to open a dialog from a dialog. I'm glad that we have validated that it is technically possible, but we already talk about it in the 'dont' section. There is no need to have an example of it. 
- [x] Does dialog title need it's own Story page, or can customizing the actions be an example on the main dialog page https://github.com/microsoft/fluentui/pull/25070

### LINKED ISSUE #25168: react-dialog: updates stories according to a11y checks
- [x] Update all examples to have close buttons, apart from the no-focusables examples
- [x] Add a docs note about NVDA reading dialog information twice, which is expected/not a bug
- [x] Add a docs external bug note about Talkback not supporting dialog name/description
- [x] Add a docs external bug note about Talkback not supporting alertdialog
- [x] Add a docs note about needing to add aria-modal="false" when including Menu, Combobox, Dropdown, or Popover within a Dialog, since otherwise VoiceOver on iOS will not be able to access the popup.

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. Updates stories accordingly to feedback provided on issues
2. Adds documentation regarding edge cases and strange behaviours on different screen readers
3. Removes nested dialogs stories and implements an e2e test to ensure we still allow the behaviour

## Related Issue(s)

Fixes #25132 and Fixes #25168

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.827 kB`<br />`17.599 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.273 kB`<br />`52.499 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `82.511 kB`<br />`24.473 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/90200f61726f12deb407611ec32067301efeb0e9'>90200f61726f12deb407611ec32067301efeb0e9</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/Avatar-mount_base.html">1287</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/Avatar-mount.html">1274</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/Button-mount_base.html">911</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/Button-mount.html">938</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProvider-mount_base.html">1498</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProvider-mount.html">1482</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProviderWithTheme-mount_base.html">570</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProviderWithTheme-mount.html">587</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">538</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">545</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">571</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/MakeStyles-mount_base.html">1952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/MakeStyles-mount.html">1942</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/SpinButton-mount_base.html">2298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25170/perf-test-react-components/SpinButton-mount.html">2289</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-dialog/src/stories/Dialog/DialogHavingAForm.stories.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
   DialogActions,
 } from '@fluentui/react-dialog';
 import { Button, Input, Label, makeStyles } from '@fluentui/react-components';
-import story from './DialogWithForm.md';
+import story from './DialogHavingAForm.md';
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/Dialog/DialogHavingAForm.stories.tsx:12]: `having a form` doesn't sound quite right to my ears 🤷

File: packages/react-components/react-dialog/e2e/DialogTitle.e2e.tsx [hunk 2]
```diff
@@ -19,18 +27,20 @@ describe('DialogTitle', () => {
             <Button>Open dialog</Button>
           </DialogTrigger>
           <DialogSurface>
-            <DialogTitle>Dialog title</DialogTitle>
             <DialogBody>
-              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam exercitationem cumque repellendus eaque
-              est dolor eius expedita nulla ullam? Tenetur reprehenderit aut voluptatum impedit voluptates in natus iure
-              cumque eaque?
+              <DialogTitle>Dialog title</DialogTitle>
```
Comment (ling1726) [inline packages/react-components/react-dialog/e2e/DialogTitle.e2e.tsx:31]: Was it just forgotten to move the title into the body ?

**Review by ling1726 — APPROVED** [review]

File: change/@fluentui-react-dialog-93e8d9f6-b34a-4eb7-8ab7-b669acc46929.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
```
Comment (ling1726) [inline change/@fluentui-react-dialog-93e8d9f6-b34a-4eb7-8ab7-b669acc46929.json:2]:
```suggestion
  "type": "none",
```

File: change/@fluentui-react-dialog-93e8d9f6-b34a-4eb7-8ab7-b669acc46929.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "chore: updates stories and tests",
+  "packageName": "@fluentui/react-dialog",
+  "email": "email not defined",
+  "dependentChangeType": "patch"
```
Comment (ling1726) [inline change/@fluentui-react-dialog-93e8d9f6-b34a-4eb7-8ab7-b669acc46929.json:6]:
```suggestion
  "dependentChangeType": "none"
```

File: packages/react-components/react-dialog/e2e/DialogTitle.e2e.tsx (hunk 2 above)
Comment (bsunderhus) [reply packages/react-components/react-dialog/e2e/DialogTitle.e2e.tsx:31]: No, it was before the introduction of `DialogContent`, so `DialogBody` moved from being a sibling of `DialogTitle` to becoming the parent, and now `DialogContent` should be the sibling.
